tests/std/tests/VSO_0000000_vector_algorithms/test.cpp

```diff
@@ -34,6 +35,66 @@ using namespace std;
 #pragma clang diagnostic ignored "-Wc++17-extensions" // constexpr if is a C++17 extension
 #endif // __clang__
 
+template <class InIt, class OutIt>
+OutIt last_known_good_adj_diff(InIt first, InIt last, OutIt dest) {
+    if (first == last) {
+        return dest;
+    }
+
+    auto val = *first;
+    *dest    = val;
+
+    for (++first, ++dest; first != last; ++first, ++dest) {
+        auto tmp = *first;
+        *dest    = tmp - val;
+        val      = tmp;
+    }
+
+    return dest;
+}
+
+template <class T>
+void test_case_adj_diff(const vector<T>& input, vector<T>& output_expected, vector<T>& output_actual) {
+    auto actual   = adjacent_difference(input.begin(), input.end(), output_actual.begin());
+    auto expected = last_known_good_adj_diff(input.begin(), input.end(), output_expected.begin());
+
+    assert(actual - output_actual.begin() == expected - output_expected.begin());
+    assert(output_actual == output_expected);
+}
+
+template <class T>
+void test_adjacent_difference(mt19937_64& gen) {
+    using Limits = numeric_limits<T>;
+
+    uniform_int_distribution<conditional_t<sizeof(T) == 1, int, T>> dis(
+        is_signed_v<T> ? static_cast<T>(Limits::min() / 2) : Limits::min(),
+        is_signed_v<T> ? static_cast<T>(Limits::max() / 2) : Limits::max());
+
+    vector<T> input;
+    vector<T> output_actual;
+    vector<T> output_expected;
+
+    vector<T>* const all__output_vectors[] = {&output_actual, &output_expected};
+    vector<T>* const all_vectors[]         = {&input, &output_actual, &output_expected};
+
+    for (auto v : all_vectors) {

```

Comment:
Style: `const auto&`, occurs below. (Yes, I irrationally loathe range-for loops that copy elements, since I tried to fix the Standard.)